### PR TITLE
image selection supporting multiple configurable options

### PR DESCRIPTION
**Pull request by mdesmet**
### Related issues

No issue

### Short description and why it's useful

If you have a shop with configurable children products having multiple configurable options that look totally different (each having different thumbnail), this code change enables showing the right product for each

### Screenshot of passed e2e tests (if you are using our standard setup as a backend)

feature not present in demo shop

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility)
- [x] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/Upgrade%20notes.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

### Contribution and currently important rules acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [x] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [x] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

### CODE REVIEW

File: core/modules/catalog/helpers/index.ts [hunk 1]
```diff
@@ -4,8 +4,10 @@ import { calculateProductTax } from '../helpers/tax'
 import flattenDeep from 'lodash-es/flattenDeep'
 import omit from 'lodash-es/omit'
 import remove from 'lodash-es/remove'
-import groupBy from 'lodash-es/groupBy'
+import reduce from 'lodash-es/reduce'
```
Comment (filrak) [inline core/modules/catalog/helpers/index.ts]: Imho we can just ES features instead of using lodash to reduce bundle size :)

Comment (mdesmet) [reply core/modules/catalog/helpers/index.ts]: Makes sense, I'll take care of it!

File: core/modules/catalog/helpers/index.ts [hunk 2]
```diff
@@ -547,25 +547,24 @@ export function getMediaGallery (product) {
...
-                })
-            }
-        })
-    } else {
-        configurableChildrenImages = attributeImages(product)
-    }
-    return configurableChildrenImages
+  if (product.configurable_children && product.configurable_children.length > 0) {
+    let configurableAttributes = product.configurable_options.map(option => option.attribute_code)
+    product.configurable_children.forEach(child => {
```
Comment (patzick) [inline core/modules/catalog/helpers/index.ts]:
```suggestion
    configurableChildrenImages  = product.configurable_children.map(child => {
```

File: core/modules/catalog/helpers/index.ts [hunk 3]
```diff
@@ -547,25 +547,24 @@ export function getMediaGallery (product) {
...
+      let id = configurableAttributes.reduce((result, attribute) => {
+        result[attribute] = child[attribute]
+        return result
+      }, {})
+      configurableChildrenImages.push({
```
Comment (patzick) [inline core/modules/catalog/helpers/index.ts]:
```suggestion
      return {
```

File: core/modules/catalog/helpers/index.ts [hunk 4]
```diff
@@ -547,25 +547,24 @@ export function getMediaGallery (product) {
...
+        'src': getThumbnailPath(product.image, rootStore.state.config.products.gallery.width, rootStore.state.config.products.gallery.height),
+        'loading': getThumbnailPath(product.image, 310, 300),
+        'error': getThumbnailPath(product.image, 310, 300),
+        'id': id
+      })
```
Comment (patzick) [inline core/modules/catalog/helpers/index.ts]:
```suggestion
      }
```

File: core/modules/catalog/helpers/index.ts (hunk 2 above)
Comment (mdesmet) [reply core/modules/catalog/helpers/index.ts]: Yes, that's definitely cleaner!

Comment (mdesmet) [reply core/modules/catalog/helpers/index.ts]: In what release will this go?

**Comment by patzick** [comment]
@mdesmet yeah, thats look a way better:) please just update changelog.md with description of your feature

**Pull request merged**